src/observer.js

```diff
@@ -349,12 +349,41 @@ function mixinLifecycleEvents(target) {
 }
 
 // TODO: support injection somehow as well?
-export const Observer = observer(({ children }) => children())
+export const Observer = observer(({ children, inject: observerInject, render }) => {
+    const component = children || render
+    if (typeof component === "undefined") {
+        return null
+    }
+    if (!observerInject) {
+        return component()
+    }
+    const InjectComponent = inject(observerInject)(component)
+    return <InjectComponent />
+})
 
 Observer.displayName = "Observer"
 
 Observer.propTypes = {
+    render: (propValue, key, componentName, location, propFullName) => {
```

Comment:
DRYing this would be nice :)

Reply:
ennnn Any suggestion to diy  Observer.propTypes

Reply:
Well, if you look closely, it's the same function except for that first condition. Which doesn't make any sense to me. It means, that when checking `render` prop, you are telling it's ok if `children` prop is a function. That doesn't make any sense to me.